### ISSUE TITLE
Separating class and its inherited class

### ISSUE THREAD

**Issue by zamiramir** [drivendataorg/erdantic#43]
As a user
I want to see the relation between a class and its inherited class and also separate the members where it should be
So that you will get a correct UML view of the classes.

example: https://github.com/drivendataorg/erdantic/blob/main/erdantic/examples/pydantic.py

Additional class:

```python
class PlannedParty(Party):
    """A planned group of adventurers finding themselves doing and saying things altogether unexpected.
    Attributes:
        name (str): Name that party is known by
        formed_datetime (datetime): Timestamp of when the party was formed
        members (List[Adventurer]): Adventurers that belong to this party
        active_quest (Optional[Quest]): Current quest that party is actively tackling
        planned_quests (List[Quest]): A list of quest that party which can be tackled
    """

    planned_quests: List[Quest] = []
```

**Comment by apirogov** [comment]
I am also very interested in being able to represent inheritance and include Base/Child classes in the diagram, and also see what data fields are narrowed down (overridden) in child classes.

@pjbull (you did the last commit so I mention you, sorry if you are not the maintainer) is this something that you see within the scope of erdantic? If so I might try to work on this (if I don't find a more suitable alternative that can do it)

**Comment by pjbull** [comment]
@apirogov Yeah, give it a shot and see where you get. I think the devil will be in the details in terms of providing users with options for how to show inheritance and when to not show it.  (E.g., in our example we don't really want `BaseModel` in our diagram or `str` or `Enum` for the `Alignment` class.

**Comment by Magier** [comment]
Chiming in to express my interest in this. 
I have a somewhat more elaborate domain model with no single root, that will span out to all other entities. If this would support an 'inverse' lookup of sorts (i.e. point to `Party` and the `PlannedParty` will also show up in the diagram) would be awesome. This would make it very user-friendly to choose an entry point for the diagram.

**Comment by jayqi** [comment]
Hello,

After considering this topic in depth, I've decided that representing inheritance relationships is out of scope for this project. I'm closing this issue as "not planned", though it remains unlocked and additional discussion is welcome. Just be aware that it will likely be a high bar to make me reconsider. 

The main points against inclusion are:

- Current diagrams are simple and easy to read because all edges uniformly represent one thing. Introducing a new type of edge will make diagrams harder to interpret. 
- Whether inherited fields should be duplicated or not across models is not obvious. This ambiguity would also make diagrams harder to read. 
- In my experience, inheritance with relationships that are semantically meaningful to visualize is not commonly a good design pattern when implementing non-trivial data models using data classes. Composition or discriminated unions are alternative design patterns that are already well-represented by erdantic.

If you are looking for a diagram that represents inheritance, take a look at UML class diagram tools like [pyreverse](https://pylint.readthedocs.io/en/stable/additional_tools/pyreverse/index.html).
